gcf/custom/manifest.go

```diff
@@ -15,109 +15,103 @@
-		ds := &pb.DataCommonsManifest_DatasetSource{
-			Url:  proto.String("https://datacommons.org/"),
-			Name: proto.String(dataSource),
-		}
-
-		ds.Datasets = make([]*pb.DataCommonsManifest_DatasetInfo, len(datasetNames))
-		for i, datasetName := range datasetNames {
-			ds.Datasets[i] = &pb.DataCommonsManifest_DatasetInfo{
-				Name: proto.String(datasetName),
-				Url:  ds.Url, // Dummy URL
-			}
-
-			for datasetName, dataFiles := range importGroupFiles.Dataset2DataFiles {
-				var tables []*pb.ExternalTable
-				tables = append(tables, &pb.ExternalTable{
-					MappingPath: proto.String(dataFiles.BigStoreTMCFPath(bucket)),
-					CsvPath:     dataFiles.BigstoreCSVPaths(bucket),
-				})
-
-				datasetImports = append(datasetImports, &pb.DataCommonsManifest_Import{
-					ImportName:               proto.String(datasetName), // Use datasetName for import name.
-					Category:                 pb.DataCommonsManifest_STATS.Enum(),
-					ProvenanceUrl:            proto.String("https://datacommons.org/"), // Dummy URL
-					McfProtoUrl:              []string{dataFiles.BigStoreMCFProtoUrl(bucket)},
-					ImportGroups:             []string{importGroupFiles.ImportGroupName},
-					ResolutionInfo:           &pb.ResolutionInfo{UsesIdResolver: proto.Bool(true)},
-					DatasetName:              proto.String(datasetName),
-					AutomatedMcfGenerationBy: proto.String(importGroupFiles.ImportGroupName),
-					Table:                    tables,
-				})
-			}
-		}
-		datasetSources = append(datasetSources, ds)
+func computeTable(bucket, root, im, tab string, table *Table) *pb.ExternalTable {
+	bigstoreTmcf := filepath.Join("/bigstore", bucket, root, "data", im, tab, table.tmcf)
+	bigstoreCsv := []string{}
+	for _, csv := range table.csv {
+		bigstoreCsv = append(
+			bigstoreCsv,
+			filepath.Join("/bigstore", bucket, root, "data", im, tab, csv),
+		)
 	}
-
-	importGroups := []*pb.DataCommonsManifest_ImportGroup{
-		{
-			Name:        proto.String(importGroupFiles.ImportGroupName),
-			IsCustomDc:  proto.Bool(true),
-			Description: proto.String("Custom DC import group"),
-		},
+	return &pb.ExternalTable{
+		MappingPath: proto.String(bigstoreTmcf),
+		CsvPath:     bigstoreCsv,
 	}
-
-	return &pb.DataCommonsManifest{
-		Import:        datasetImports,
-		DatasetSource: datasetSources,
-		ImportGroups:  importGroups,
-	}, nil
 }
 
-// pathToDataFolder is the "gs://"-prefixed folder that contains raw data.
-// For the folder structure that is expected, please see,
-// https://docs.datacommons.org/custom_dc/upload_data.html
-func GenerateManifest(
-	ctx context.Context,
-	bucket, dataDir string,
+func ComputeManifest(
+	bucket string,
+	layout *Layout,
 ) (*pb.DataCommonsManifest, error) {
-	client, err := storage.NewClient(ctx)
-	if err != nil {
-		return nil, err
+	root := layout.root
+	importGroup := path.Base(root)
+	// IMPORTANT NOTE:
+	// importGroup has a character constraint of 21 due to internal limitations.
```

Comment:
It says "21" but the code below says 20?

Reply:
changed comments to 20